src/algorithms/tracking/CKFTracking.cc

```diff
@@ -371,6 +451,11 @@ namespace eicrecon {
 
           lastSeed = constSeedNumber(track);
 
+          if (!track.hasReferenceSurface()) {
+            ACTS_WARNING("Track has no reference surface.");
+            continue;
+          }
```

Comment:
Where did this come from? Upstream code only has a `ACTS_ERROR("Track has no reference surface");` but in a different context.

Reply:
Also, to simply continue here is not correct. We tried that before, and this is what was implemented instead: https://github.com/eic/EICrecon/pull/1677

Reply:
Thanks for the careful review. Segfaults happen two lines below this even with the #1677 treatment included (which has always been in this branch for testing since it well predates it). I guess that means even with a successful extrapolation we now have referenceSurfacePtrs that are null and the check in #1677 is insufficient.

Suggestion: revert this, and I'll dig into it more.

Reply:
Actually, it may be that in #1677 we should have had the addition to failed tracks also in the smoothing step.

Reply:
At least this is an easy to reproduce issue, since it segfaults out on the very first event (and presumably every event).

Reply:
#1741 doesn't address the underlying issue, but is a sensible fix anyway.